addons/mrp/__init__.py

```diff
@@ -7,6 +7,25 @@
 
 from odoo import api, SUPERUSER_ID
 
+
+def _pre_init_mrp(cr):
+    """ Allow installing MRP in databases with large stock.move / stock.move.line tables (>1M records)
+        - Creating the computed+stored field stock_move.is_done is terribly slow with the ORM and
+          leads to "Out of Memory" crashes
+        - stock.move.line.done_move is a stored+related on the former... """
+    cr.execute("""ALTER TABLE "stock_move" ADD COLUMN "unit_factor" float;""")
+    cr.execute("""ALTER TABLE "stock_move" ADD COLUMN "is_done" bool;""")
+    cr.execute("""ALTER TABLE "stock_move_line" ADD COLUMN "done_move" bool;""")
+    cr.execute("""UPDATE stock_move
+                     SET is_done=TRUE,
+                         unit_factor=1.0
+                   WHERE STATE='done'
+                      OR STATE='cancel';""")
```

Comment:
To update all rows, you should do
```suggestion
    cr.execute("""UPDATE stock_move
                     SET is_done=COALESCE(state IN ('done', 'cancel'), false),
                         unit_factor=1.0""")
```

Reply:
Agreed :)